### DISCUSSION TITLE
Permission denied on userscripts

### DISCUSSION THREAD

**Discussion by gfsd3v** [qutebrowser/qutebrowser#5838] [Q&A] [answered]
Hello guys.

I'm having some trouble trying to execute the qute-bitwarden user scripts, heres the log:

`17:45:06 DEBUG    procs      guiprocess:_pre_start:151 Executing: /home/gfs/.local/share/qutebrowser/userscripts/qute-bitwarden
17:45:06 DEBUG    message    message:_log_stack:42 Stack for error message:
  File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/gfs/repos/qutebrowser/qutebrowser/__main__.py", line 29, in <module>
    sys.exit(qutebrowser.qutebrowser.main())
  File "/home/gfs/repos/qutebrowser/qutebrowser/qutebrowser.py", line 204, in main
    return app.run(args)
  File "/home/gfs/repos/qutebrowser/qutebrowser/app.py", line 128, in run
    ret = qt_mainloop()
  File "/home/gfs/repos/qutebrowser/qutebrowser/app.py", line 138, in qt_mainloop
    return q_app.exec_()
  File "/home/gfs/repos/qutebrowser/qutebrowser/misc/guiprocess.py", line 82, in _on_error
    message.error("Error while spawning {}: {}".format(self._what, msg))
  File "/home/gfs/repos/qutebrowser/qutebrowser/utils/message.py", line 54, in error
    stack = ''.join(traceback.format_stack())
17:45:06 ERROR    message    message:error:59 Error while spawning userscript: execve: Permission denied`

![image](https://user-images.githubusercontent.com/36668882/97227273-0297ff80-17b4-11eb-8a59-d7cd93158615.png)

It was working before I deleted my "~/.local/share/qutebrowser" folder after some problems, does anyone has any idea on how to fix it ?

System: Ubtuntu 20.04
Installed via python virtualenv

**Comment by The-Compiler** [comment]
What does `ls -l /home/gfs/.local/share/qutebrowser/userscripts/qute-bitwarden` say?

**Reply by gfsd3v** [reply]
> ls -l /home/gfs/.local/share/qutebrowser/userscripts/qute-bitwarden

-rw-rw-r-- 1 gfs gfs 10707 Oct 26 14:46 /home/gfs/.local/share/qutebrowser/userscripts/qute-bitwarden

**Comment by The-Compiler** [comment, marked as answer]
From the output above, your script doesn't have the executable bit set, thus resulting in a "permission denied" error when trying to execute it. A `chmod +x ~/.local/share/qutebrowser/userscripts/qute-bitwarden` (and perhaps also other userscripts in that folder) should help.

**Reply by gfsd3v** [reply]
This fixed, thanks. 

I tried a "chmod +x ~/.local/share/qutebrowser/userscripts" before thinking that it would do a recursive strategy on all files, sorry for the trouble and thank you very much.

**Reply by The-Compiler** [reply]
You're welcome! :slightly_smiling_face: The command above will set the executable bit on the directory, which is already set by default and controls whether you can access content of the directory (with the `r` bit set, you can only see the file list, but not `cd` into it or access the files).

Thus, your command above needs a `chmod -R` (`--recursive`) or `~/.local/share/qutebrowser/userscripts/*` - I'd prefer the latter, as I try to avoid `chmod -R` as it can be quite destructive when accidentally run on the wrong folder.

**Reply by gfsd3v** [reply]
Thanks for explaining and for the tips, It will surely help a lot.
